backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/IADWindowDAO.java

```diff
@@ -41,6 +42,8 @@ public interface IADWindowDAO extends ISingletonService
 
 	AdWindowId getWindowIdByInternalName(String internalName);
 
+	Optional<AdWindowId> getOverridingWindowId(AdWindowId adWindowId);
```

Comment:
this is not needed, pls read below

Reply:
wherever you need to get the actual window ID, pls use `de.metas.document.references.zoom_into.CustomizedWindowInfoMapRepository`

e.g.
```java
		return customizedWindowInfoMapRepository.get()
				.getCustomizedWindowInfo(adWindowId)
				.map(CustomizedWindowInfo::getCustomizationWindowId)
				.orElse(adWindowId);
```

That code snippet will give you the effective window ID to be used.

Reply:
=> https://github.com/metasfresh/metasfresh/pull/13514